packages/injected/src/ariaSnapshot.ts

```diff
@@ -72,11 +83,12 @@ export function generateAriaTree(rootElement: Element, options?: { forAI?: boole
       return;
 
     const element = node as Element;
-    let isVisible = !roleUtils.isElementHiddenForAria(element);
-    if (options?.forAI)
-      isVisible = isVisible || isElementVisible(element);
-    if (!isVisible)
+    if (!isVisible(element, options)) {
+      if (options?.forAI)
+        // skip this element, but still process its children https://github.com/microsoft/playwright/issues/36296
+        processElement(ariaNode, element, []);
```

Comment:
We should be consistent with the aria-owns children. We either respect them in forAI or we don't. Your change makes it inconsistent.

Reply:
Good point, didn't think about aria-owns. Moving this new logic into `toAriaNode` made this a lot better.